web_src/js/features/repo-issue.js

```diff
@@ -87,21 +87,55 @@ export function initRepoIssueDue() {
 }
 
 export function initRepoIssueSidebarList() {
+  initIssueSearchDropdown('#new-dependency-drop-list');
+
+  function excludeLabel(item) {
+    const href = $(item).attr('href');
+    const id = $(item).data('label-id');
+
+    const regStr = `labels=((?:-?[0-9]+%2c)*)(${id})((?:%2c-?[0-9]+)*)&`;
+    const newStr = 'labels=$1-$2$3&';
+
+    window.location = href.replace(new RegExp(regStr), newStr);
+  }
```

Comment:
Move function to top-level scope, and ideally export it and add some tests, e.g. pass in `href` and `labelId` and make it return `href` to make it easily testable.

Reply:
I have added a test for `excludeLabel()` (edd9408 and ded55c9). However, when I execute `make test-frontend`, it will return a error: `Error: Failed to resolve entry for package "monaco-editor". The package may have incorrect main/module/exports specified in its package.json.` Maybe it is related to this `import`:  https://github.com/go-gitea/gitea/blob/0fd1672ae49a5f69fca7d90336ae75be83a21014/web_src/js/features/repo-issue.js#L6

Then I did a search and found this [issue](https://github.com/vitest-dev/vitest/discussions/1806), could you help me take a look.

Reply:
Doing something specific in vitest config is safe as it has no effect on the prod build. I will try to verify this workaround.

Reply:
> Does the error reproduce on CI as well?

yes it does

Reply:
lint is failed: https://github.com/go-gitea/gitea/actions/runs/5643818281/job/15286390105?pr=25362.
I can't handle this, could you help me?

Reply:
> lint is failed

fixed...